src/Illuminate/Database/Grammar.php

```diff
@@ -208,14 +208,19 @@ public function isExpression($value)
     }
 
     /**
-     * Get the value of a raw expression.
+     * Transforms expressions to their scalar types.
      *
-     * @param  \Illuminate\Database\Query\Expression  $expression
-     * @return mixed
+     * @param  scalar|\Illuminate\Contracts\Database\Query\Expression  $expression
+     * @return scalar
```

Comment:
This doesn't seem correct. I can't think of any place where we used `scalar` before. Please add the correct types like int, string, or whatever's needed here.

Reply:
The correct type is now `int|float|string|bool` because the expression can transform these values to SQL. And that is exactly the definition of `scalar`. Do you want me to change it nevertheless?

Reply:
Yes please use those.

Reply:
Done